iroh-net/src/discovery/pkarr.rs

```diff
@@ -177,15 +178,16 @@ impl PkarrPublisher {
     /// This uses the pkarr relay server operated by [number 0], at
     /// [`N0_DNS_PKARR_RELAY_PROD`].
     ///
-    /// When compiling for tests, i.e. when `cfg(test)` is true, or when the `test-utils`
-    /// crate feature is enabled the [`N0_DNS_PKARR_RELAY_STAGING`] server is used instead.
+    /// When running with the environment variable
+    /// `IROH_FORCE_STAGING_RELAYS` set to any non empty value [`N0_DNS_PKARR_RELAY_STAGING`]
+    /// server is used instead.
     ///
     /// [number 0]: https://n0.computer
     pub fn n0_dns(secret_key: SecretKey) -> Self {
-        #[cfg(not(any(test, feature = "test-utils")))]
-        let pkarr_relay = N0_DNS_PKARR_RELAY_PROD;
-        #[cfg(any(test, feature = "test-utils"))]
-        let pkarr_relay = N0_DNS_PKARR_RELAY_STAGING;
+        let pkarr_relay = match std::env::var(ENV_FORCE_STAGING_RELAYS) {
+            Ok(value) if !value.is_empty() => N0_DNS_PKARR_RELAY_STAGING,
+            _ => N0_DNS_PKARR_RELAY_PROD,
+        };
```

Comment:
This should probably also do the `cfg(test)` dance as the other things, do, right?

Reply:
Hmm no, I think `TEST_DNS_NODE_ORIGIN` is a special case: It uses the `.test` TLD which has special meaning in DNS resolvers!

We don't have any equivalent for any of the other infra identifiers.

Reply:
So from my recollection `TEST_DNS_NODE_ORIGIN` was added so that it uses a special URL which will not be resolved by any normal resolvers and will only be resolved if you start a local resolver in the test using a helper function.  This stops tests from hitting any external DNS server.

Reply:
IIRC we basically didn't have any requests and this was more a precaution when we were dealing with the relays as well. (Though in spirit I do agree with it)